### DISCUSSION TITLE
Ability to publish events when "action" is partly set

### DISCUSSION THREAD

**Discussion by kinosang** [HASwitchPlate/openHASP#796] [Ideas]
For example, a button has `"action":{"up":"page 2"}` should publish `down`,`release`,`long`,`hold`,`changed` instead of omit them.

Or add command to publish MQTT message

**Comment by RyanEwen** [comment]
Came here to request both of these. I spent way too much time trying to react to events in HA for a button that had an action defined already, and since that doesn't work, the next workaround would be to issue MQTT commands right from an action defined in the JSONL. 

Context: I want my prev/home/next buttons to not only change pages but do other things in HA such as reload config

**Comment by maimrn** [comment]
Small displays are usually used in projects, and losing the functionality of publishing MQTT commands/events on objects that have "action" enabled is very bad scenario.

**Comment by AreaKode** [comment]
Just ran into this issue today. I'm using "action" to make my elements respond better to button presses. But then I don't get an MQTT message telling me that the element has been pressed.
